storage-operator/pkg/apis/storage/v1alpha1/volume_types.go

```diff
@@ -42,12 +48,88 @@ type VolumeSpec struct {
 	VolumeSource `json:",inline"`
 }
 
+type PreparationStep string
+
+// "Enum" representing the preparation steps of a volume.
+const (
+	PreparationStarted PreparationStep = "PreparationStarted"
+	PrepareVolume      PreparationStep = "PrepareVolume"
+	GetVolumeSize      PreparationStep = "GetVolumeSize"
+	CreatePV           PreparationStep = "CreatePersistentVolume"
+)
+
+type FinalizationStep string
+
+// "Enum" representing the preparation steps of a volume.
+const (
+	FinalizationStarted FinalizationStep = "FinalizationStarted"
+	UnprepareVolume     FinalizationStep = "UnprepareVolume"
+	DeletePV            FinalizationStep = "DeletePersistentVolume"
+)
+
+type VolumePhase string
+
+// "Enum" representing the preparation steps of a volume.
+const (
+	VolumeAvailable   VolumePhase = "Available"
+	VolumePending     VolumePhase = "Pending"
+	VolumeFailed      VolumePhase = "Failed"
+	VolumeTerminating VolumePhase = "Terminating"
+)
+
+type VolumeErrorCode int
+
+// "Enum" representing the error codes of the Failed state.
```

Comment:
In general, the K8s ecosystem seems to prefer well-known/constant strings over numbers for various kinds of codes. That way, `kubectl get ...` and such provide much better output for a human, and a machine doesn't really care whether you hard-code numbers or hard-code strings anyway. I'd argue following said approach is good practice.

Reply:
Makes sense, I'll use string.